Xamarin.Forms.Platform.Android/FastRenderers/AccessibilityProvider.cs

```diff
@@ -137,19 +136,15 @@ bool SetHint(string hint = GetFromElement)
 				return true;
 			}
 
-			if (_defaultHint == null)
-			{
-				_defaultHint = textView.Hint;
-			}
+			string accessibilityName = (string)Element.GetValue(Accessibility.NameProperty);
+			string accessibilityHint = (string)Element.GetValue(Accessibility.HintProperty);
 
-			string value = hint;
-			if (value == GetFromElement)
+			if (string.IsNullOrWhiteSpace(accessibilityName) && string.IsNullOrWhiteSpace(accessibilityHint))
 			{
-				value = string.Join(". ", (string)Element.GetValue(Accessibility.NameProperty),
-					(string)Element.GetValue(Accessibility.HintProperty));
+				return true;
 			}
 
-			textView.Hint = !string.IsNullOrWhiteSpace(value) ? value : _defaultHint;
+			textView.Hint = string.Join(". ", accessibilityName, accessibilityHint);
```

Comment:
I no longer understand this PR.
If the name or the hint is null, do you still expect the ". " ?
Also, using string.Join is probably serious overkill